core/src/main/java/org/mskcc/oncotree/topbraid/TopBraidRepository.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2017 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.  The software and
+ * documentation provided hereunder is on an "as is" basis, and
+ * Memorial Sloan-Kettering Cancer Center
+ * has no obligations to provide maintenance, support,
+ * updates, enhancements or modifications.  In no event shall
+ * Memorial Sloan-Kettering Cancer Center
+ * be liable to any party for direct, indirect, special,
+ * incidental or consequential damages, including lost profits, arising
+ * out of the use of this software and its documentation, even if
+ * Memorial Sloan-Kettering Cancer Center
+ * has been advised of the possibility of such damage.
+*/
+
+package org.mskcc.oncotree.topbraid;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.apache.log4j.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ *
+ * @author Manda Wilson
+ **/
+@Repository
+public abstract class TopBraidRepository<T> {
+
+    private final static Logger logger = Logger.getLogger(TopBraidRepository.class);
+
+    @Value("${topbraid.url}")
+    private String topBraidURL;
+
+    @Autowired
+    private TopBraidSessionConfiguration topBraidSessionConfiguration; 
+
+    protected List<T> query(String username, String password, String query, ParameterizedTypeReference parameterizedType) {
```

Comment:
Here we are using the topbraid.url string through spring value setting, but passing in the username and password as caller parameters. I'm not sure why we use different mechanisms to pass in config info.